src/util.py

```diff
@@ -260,12 +265,111 @@ def xr_add_cyclic_longitudes(da, coord):
 
     return new_da
 
+def set_vector_density(data, lat_density=1, lon_density=1, minDistance=0, otherVars=[]):
+    """
+    Utility function to change density of vector plots.
+
+    Ars:
+
+        data (:class:`xarray.core.dataarray.DataArray`):
+            Data array that contains the vector plot latitude/longitude data.
+
+        lat_density (:class:`int`):
+            Value in range (0,1] that determines the density of the vectors in the y range.
+
+        lon_density (:class:`int`):
+            Value in range (0,1] that determines the density of the vectors in the x range.
+
+        minDistance (:class:`int`):
+            Value in degrees that determines the distance between the vectors.
+        
+        otherVars (:class: array of `xarray.core.dataarray.DataArray`):
+            Other variables that require the same density reduction as the lat/lon data
+
+    """
+    import math
+
+    if minDistance != 0 and (lat_density != 1 or lon_density != 1):
+        raise Exception("minDistance and lat/lon_density parameters cannot be used simulatenously.")
+
+    # Change the density with parameter "minDistance"
+    if minDistance != 0:
+
+        lat_every = 1
+        lon_every = 1
+
+        # Get distance between points in latitude (y axis)
+        lat = data['lat']
+        latdifference = (float)(lat[1] - lat[0])
+
+        # Get distance between points in longitude (x axis)
+        lon = data['lon']
+        londifference = (float)(lon[1] - lon[0])
+
+        # Get distance between points that are diagonally adjacent
+        diagDifference = math.sqrt(latdifference**2 + londifference**2)
+
+        # Initialize ds
+        ds = data.isel(lat=slice(None, None, None), lon=slice(None, None, None))
```

Comment:
I think in addition to initializing `ds` to fix that bug, a warning should also be thrown when the `minDistance` is shorter than the spatial resolution of the data. That way, users know why their plot isn't changing when they specify a smaller `minDistance`.

Reply:
good idea! I'll add that in.